pkg/indexers/util.go

```diff
@@ -43,3 +43,18 @@ func AppendOrDie(indexers ...cache.Indexers) cache.Indexers {
 	}
 	return ret
 }
+
+// AddIfNotPresentOrDie tries to add everything from toAdd to indexer's indexers that does not already exist. It panics
+// if it encounters an error.
+func AddIfNotPresentOrDie(indexer cache.Indexer, toAdd cache.Indexers) {
+	existing := indexer.GetIndexers()
+	for indexName := range toAdd {
+		if _, exists := existing[indexName]; exists {
+			delete(toAdd, indexName)
```

Comment:
a pity function pointers cannot be compared

Reply:
This may seem silly, but this would cleanly allow during startup two indexers named the same but do two different things, right?

I feel like a bug like that would be hard to track down. Would it be worth the cost of using reflection to compare the function pointers?

https://pkg.go.dev/reflect#Value.Pointer